src/core/gltf2exporter/gltf2exporter_p.h

```diff
@@ -0,0 +1,115 @@
+/*
+    gltf2exporter.h
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Jean-Michaël Celerier <jean-michael.celerier@kdab.com>
+
+    Licensees holding valid proprietary KDAB Kuesa licenses may use this file in
+    accordance with the Kuesa Enterprise License Agreement provided with the Software in the
+    LICENSE.KUESA.ENTERPRISE file.
+
+    Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef GLTF2EXPORTER_H
+#define GLTF2EXPORTER_H
+
+#include <QObject>
+#include <QUrl>
+#include <QDir>
+#include <QJsonObject>
+#include <QMap>
+#include <Kuesa/kuesa_global.h>
+
+QT_BEGIN_NAMESPACE
+
+namespace Kuesa {
+class GLTF2Importer;
+namespace GLTF2Import {
+class GLTF2Context;
+}
+class SceneEntity;
+
+class KUESASHARED_EXPORT GLTF2ExportConfiguration
+{
+public:
+    enum MeshAttribute { Position,
+                         Normal,
+                         Color,
+                         TextureCoordinate,
+                         Generic };
+
+    // O = slowest, 10 = fastest
+    void setMeshEncodingSpeed(int speed);
+    int meshEncodingSpeed() const;
+    void setMeshDecodingSpeed(int speed);
+    int meshDecodingSpeed() const;
+
+    void setAttributeQuantizationLevel(MeshAttribute attribute, int level);
+    int attributeQuantizationLevel(MeshAttribute) const;
+
+    void setMeshCompressionEnabled(bool enabled);
+    bool meshCompressionEnabled() const;
+
+private:
+    int m_encodingSpeed{};
+    int m_decodingSpeed{};
+    bool m_meshCompression{};
+    QMap<MeshAttribute, int> m_quantization;
+};
+
+class KUESASHARED_EXPORT GLTF2Exporter : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(SceneEntity *scene READ scene WRITE setScene NOTIFY sceneChanged)
+
+public:
+    explicit GLTF2Exporter(QObject *parent = nullptr);
+
+    SceneEntity *scene() const;
+    QStringList errors() const;
+
+    GLTF2ExportConfiguration configuration() const;
+    void setConfiguration(const GLTF2ExportConfiguration &conf);
+
+    void setContext(GLTF2Import::GLTF2Context *context);
```

Comment:
Context is private API so it's unlikely user code will be able to put one here (or we force them to do use kuesa-private). Are we sure this is required?
Ditto for the following method, this context class really feels like an implementation detail leaking through API.

Reply:
hmm, it is "only" needed to access the structures defined in meshparser_p.h - I don't see other easy ways right now (short of re-parsing the original gltf inside the gltfexporter, recreating the qt3d nodes, and operating on those instead).

Reply:
another solution would be to rewrite the dracocompressor file so that it isn't based on qt3d entities / attributes / etc. However this will break the day where the scene can be modified / combined / whatever, and since it is on the roadmap...

Reply:
well GLTF2Exporter is also private so what's the issue?

Reply:
Oh right, I was confused by the export macro... But indeed for some reason we don't have a KUESA_PRIVATE_EXPORT macro. Maybe that should be changed? (in a separate commit)

Reply:
Now that it's introduced (thanks btw!), maybe it's time to start using it here?